auth/forward.go

```diff
@@ -0,0 +1,88 @@
+package auth
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/types"
+	"github.com/stretchr/stew/objects"
+)
+
+// Forward the authentication to a external server
+func Forward(forward *types.Forward, w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+
+	client := http.Client{}
+
+	forwardReq, err := http.NewRequest("GET", forward.Address, nil)
+	if err != nil {
+		log.Debugf("Error calling %s. Cause %s", forward.Address, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	forwardReq.Header.Add("Accept", "application/json")
+	forwardQuery := forwardReq.URL.Query()
+	rQuery := r.URL.Query()
+	for _, reqParam := range forward.RequestParameters {
+		paramValue := rQuery.Get(reqParam.Name)
+		forwardQuery.Add(reqParam.As, paramValue)
+	}
+	forwardReq.URL.RawQuery = forwardQuery.Encode()
+	forwardResponse, forwardErr := client.Do(forwardReq)
+	if forwardErr != nil {
+		log.Debugf("Error calling %s. Cause: %s", forward.Address, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	defer forwardResponse.Body.Close()
+	body, readError := ioutil.ReadAll(forwardResponse.Body)
+	if readError != nil {
+		log.Debugf("Error reading body %s. Cause: %s", forward.Address, readError)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if forwardResponse.StatusCode != 200 {
+		log.Debugf("Remote error %s. StatusCode: %s", forward.Address, forwardResponse.StatusCode)
+		w.WriteHeader(forwardResponse.StatusCode)
+		w.Write(body)
```

Comment:
Should probably return after this line, otherwise it continues trying to parse the JSON response even though there was an error.